stunnel/stunnel_cache.ml

```diff
@@ -191,6 +199,13 @@ let connect ?use_fork_exec_helper ?write_to_log host port verify_cert =
   try
     remove host port verify_cert
   with Not_found ->
-    error "Failed to find stunnel in cache for endpoint %s:%d" host port;
+    info "connect did not find cached stunnel for endpoint %s:%d" host port;
     Stunnel.connect ?use_fork_exec_helper ?write_to_log ~verify_cert host port
     
+let set_legacy_protocol_and_ciphersuites_allowed b =
+	if b <> !Stunnel.legacy_protocol_and_ciphersuites_allowed then (
```

Comment:
Is this thread-safe? We discussed whether it might be better to not bother flushing since any inappropriate connections are removed from the cache on both cache-add and cache-lookup. This would mean we needn't leak as much in the `stunnel.mli`.